src/components/CommitmentDetails.tsx

```diff
@@ -0,0 +1,251 @@
+import React from 'react';
+
+import { useGet, CommitmentsQueryParams } from '../hooks/axiosHooks';
+
+import fixCasing from '../utils/fixCasing';
+
+import theme from '../theme';
+
+import {
+  createMuiTheme,
+  ThemeProvider,
+  makeStyles,
+} from '@material-ui/core/styles';
+import { Divider, Button } from '@material-ui/core';
+import CheckCircleOutlineIcon from '@material-ui/icons/CheckCircleOutline';
+import AccountCircleIcon from '@material-ui/icons/AccountCircle';
+import PaymentIcon from '@material-ui/icons/Payment';
+
+import PaymentDetails from './PaymentDetails';
+import RecurringPayments from './RecurringPayments';
+
+const MUITheme = createMuiTheme(theme);
+
+export interface DetailProps {
+  displayId: string;
+  setDisplayId: (displayId: string) => void;
+}
+
+const useStyles = makeStyles(theme => {
+  return {
+    errorText: {
+      color: theme.palette.error.main,
+    },
+    flexRow: {
+      display: 'flex',
+      flexDirection: 'row',
+    },
+    pageTitle: {
+      textAlign: 'left',
+    },
+    iconContainer: {
+      display: 'flex',
+      alignItems: 'center',
+    },
+    paymentContainer: {
+      display: 'flex',
+      flexDirection: 'row',
+      maxWidth: '400px',
+      minWidth: '35%',
+      justifyContent: 'space-between',
+      alignItems: 'center',
+      marginBottom: '25px',
+    },
+    userContainer: {
+      display: 'flex',
+      flexDirection: 'row',
+      maxWidth: '50%',
+      minWidth: '500px',
+      textAlign: 'left',
+      marginBottom: '20px',
+      alignItems: 'center',
+    },
+    infoContainer: {
+      display: 'flex',
+      flexDirection: 'column',
+    },
+    cardContainer: {
+      display: 'flex',
+      flexDirection: 'row',
+      justifyContent: 'space-between',
+      minWidth: '185px',
+      marginLeft: '13px',
+      marginBottom: '17px',
+    },
+    textBold: {
+      fontWeight: 'bold',
+    },
+    textLight: {
+      fontWeight: 'lighter',
+    },
+    checkCircle: {
+      color: '#14FF52',
+      fontSize: 'medium',
+      marginRight: '5px',
+    },
+    accountIcon: {
+      fontSize: 'medium',
+      marginBottom: '18px',
+      marginRight: '6px',
+    },
+    paymentIcon: {
+      fontSize: 'medium',
+    },
+
+    titleContainer: {
+      display: 'flex',
+      flexDirection: 'row',
+      justifyContent: 'space-between',
+    },
+    cancelButton: {
+      height: '50%',
+    },
+  };
+});
+
+const CommitmentDetails = (props: DetailProps): JSX.Element => {
+  const classes = useStyles(theme);
+  const { displayId, setDisplayId } = props;
+
+  const queryParams: CommitmentsQueryParams = {
+    limit: 1,
+    page: 0,
+  };
+
+  const [response, loading, error] = useGet(
+    `commitment/${displayId}`,
+    queryParams
+  );
+
+  if (loading) {
+    console.log('loading');
+    return <div></div>;
+  }
+  const data = response && response.data; // same response?.data
+
+  if (!data) {
+    return <div></div>;
+  }
```

Comment:
You don't have a case for `error` on this page, so it renders as blank if, for example, you try to load the page with the mock server off.  I'd also recommend taking out the console log and putting `loading...` into the div you return.